march_gait_selection/src/march_gait_selection/transition_gait/create_transition.py

```diff
@@ -0,0 +1,206 @@
+                            .format(cg=old_gait_name, ng=new_gait_name))
+
+        old_subgait_name = new_subgait_name if old_subgait_name is None else old_subgait_name
+
+        old_subgait = old_gait[old_subgait_name]
+        new_subgait = new_gait[new_subgait_name]
+
+        if old_subgait is None:
+            raise SubgaitNameNotFound(subgait_name=old_subgait_name)
+
+        if new_subgait is None:
+            raise SubgaitNameNotFound(subgait_name=new_subgait_name)
+
+        old_subgait, new_subgait = self._scale_subgait_duration(old_subgait, new_subgait)
+
+        joints, transition_positions, transition_velocities, transition_durations = \
+            self._transition_trajectory(old_subgait, new_subgait)
+
+        subgait_dict = self._to_subgait_dictionary(joints, transition_positions,
+                                                   transition_velocities, transition_durations)
+
+        return self._new_subgait(self._robot, subgait_dict)
+
+    @staticmethod
+    def _scale_subgait_duration(old_subgait, new_subgait):
+        """Scale the subgaits to have matching setpoints and corresponding timestamps.
+
+        :param old_subgait:
+            The subgait which the transition is starting from
+        :param new_subgait:
+            The subgait which the transition will result in.
+
+        :return:
+            Same subgaits but with matching amount of setpoints and matching timestamps
+        """
+        timestamps = sorted(set(old_subgait.get_unique_timestamps() + new_subgait.get_unique_timestamps()))
+
+        for old_joint in old_subgait.joints:
+            new_joint = new_subgait.get_joint(old_joint.name)
+
+            old_joint_setpoints = []
+            new_joint_setpoints = []
+            for timestamp in timestamps:
+                old_joint_setpoints.append(old_joint.get_interpolated_setpoint(timestamp))
+                new_joint_setpoints.append(new_joint.get_interpolated_setpoint(timestamp))
+
+            old_joint.setpoints = old_joint_setpoints
+            new_joint.setpoints = new_joint_setpoints
+
+        return old_subgait, new_subgait
+
+    @staticmethod
+    def _transition_trajectory(old_subgait, new_subgait):
+        """Calculate a transition trajectory which starts at the old gait and ends with the endpoints of the new gait.
+
+        :param old_subgait:
+            The subgait which the transition is starting from
+        :param new_subgait:
+            The subgait which the transition will result in.
+
+        :returns:
+             list of joints, velocities and positions corresponding to every joint at given timestamps and duration
+        """
+        joints = []
+        transition_positions = []
+        transition_velocities = []
+        transition_durations = []
+
+        total_transition_factor = len(new_subgait[0].setpoints)
+        for transition_index in range(total_transition_factor):
```

Comment:
```suggestion
        for setpoint_index in range(number_setpoints):
```
I think it will be more intuitive if you name them after what they are rather than what they are used for